### ISSUE TITLE
Approach state of art

### ISSUE THREAD

**Issue by arjunmenon** [jekyll/classifier-reborn#158]
How do you include the TF-IDF weights in this method?
Compared to simple MNB having count of plain bag of words, MNB with TF-IDF gets more accuracy.
How do you implement this?
www.cs.waikato.ac.nz/~eibe/pubs/kibriya_et_al_cr.pdf

Also, as a bonus a complement naive bayes further improves the problem of inconsistent dataset size which with plain MNB favours the larger one.
This scenario is fairly common.
Some inputs on that as well would be appreciated,.

**Comment by Ch4s3** [comment]
We've discussed TF-IDF, but I haven't had time to dig into it.

**Comment by cardmagic** [comment]
FYI, the original [classifier gem](https://github.com/cardmagic/classifier) now has TF-IDF built in as a standalone vectorizer class.

```ruby
require 'classifier'

# Create a TF-IDF vectorizer with optional configuration
tfidf = Classifier::TFIDF.new(
  min_df: 2,           # Ignore terms appearing in fewer than 2 documents
  max_df: 0.95,        # Ignore terms appearing in >95% of documents
  ngram_range: [1, 2], # Unigrams and bigrams
  sublinear_tf: true   # Use 1 + log(tf) instead of raw tf
)

# Fit on corpus
tfidf.fit([
  'Dogs are great pets',
  'Cats are independent animals',
  'Dogs and cats can be friends'
])

# Transform documents to TF-IDF vectors
vector = tfidf.transform('Dogs are loyal companions')
# => {:dog=>0.7071..., :loyal=>0.7071...}

# Or use fit_transform for both steps
vectors = tfidf.fit_transform(documents)
```

It also supports:
- Persistence via the pluggable storage system (`save`, `load`, JSON serialization)
- Streaming from files via `fit_from_stream` for large corpora
- N-gram support (unigrams, bigrams, etc.)

See: https://github.com/cardmagic/classifier/blob/master/lib/classifier/tfidf.rb
